Clarify subsection ID syntax in assemblies

adellape opened the pull request:
* Per team discussion, add clarifications around using `{context}` variables in assembly _subsection_ anchor IDs ([Preview](https://github.com/adellape/openshift-docs/blob/guidelines_1/contributing_to_docs/doc_guidelines.adoc#anchoring-in-assembly-files)). That is, _do_ use them for subsections, but not for the main assembly title/heading.
  * Also updates the ["Anchoring "Prerequisites", "Additional resources", and "Next steps" titles in assemblies"](https://github.com/adellape/openshift-docs/blob/guidelines_1/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies) section to match the use of `_{context}`.
* Peppers in language specific to Level 0, Level 1, etc per https://docs.asciidoctor.org/asciidoc/latest/sections/titles-and-levels/#section-level-syntax 
* Reflowing of the "Assembly/module titles and section headings" section for clarity, most notably moving "Discrete headings" as the last subsection:
<img width="411" alt="image" src="https://github.com/openshift/openshift-docs/assets/3442316/a5cfc121-72c7-46ad-b9a1-022ea0abe5aa">

* Update some older example blocks that were using either manually-bolded titles or an unnecessary lead-in section to just use a `.Example` style block title

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -266,41 +266,23 @@ Do not use special characters or symbols in titles. Symbols and special characte
 = Configuring alert notifications
 ----
 
-[NOTE]
-====
 Do not include line spaces between the anchor ID and the section title.
-====
```
Didn't seem like this needed to be a NOTE when sections right before it just had line-by-line "Do not...", "Do not...", etc. (i.e., this one didn't seem more special).

JoeAldinger commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -334,28 +314,49 @@ The `prerequisites_`, `additional-resources_`, and `next-steps_` prefixes must e
 
+=== Discrete headings
+
+If you have a section heading that you do not want to appear in the TOC (like if you think that some section is not worth showing up or if there are already too many nested levels), you can use a discrete (or floating) heading:
```
```suggestion
If you have a section heading that you do not want to appear in the TOC (for example, if you think that some section is not worth showing up or if there are already too many nested levels), you can use a discrete (or floating) heading:
```

JoeAldinger replied:
I don't think this necessarily needs to be changed just a suggestion to get rid of `like` here.

JoeAldinger reviewed (approved):
/lgtm

adellape commented:
Putting this through peer review again because it has had some significant changes since the last round, not least of which flipping on what the actual advice is. 😅

skopacz1 reviewed (commented):
Left some comments, otherwise LGTM! Glad to see this finally moving forward

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -287,98 +287,137 @@ Do not use backticks or other markup in assembly or module headings.
+===== Assembly subsections (Level 1 and lower headings)
 
-You must add the `{context}` variable to the end of each anchor ID in module files. When called, the `{context}` variable is resolved into the value declared in the `:context:` attribute in the corresponding assembly file. This enables cross-referencing to module IDs in context to a specific assembly and is useful when a module is included in multiple assemblies.
+For any Level 1 or lower _subsections_ in an assembly (for example, `==`, `===`, etc.), add a `_{context}` variable to the end of anchor IDs in assembly section headings of any level (`=`, `==`, etc.).
```
Oops yea, was modifying the earlier version of the PR's sentence and left garbage.

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -287,98 +287,137 @@ Do not use backticks or other markup in assembly or module headings.
+
+https://docs.asciidoctor.org/asciidoc/latest/blocks/discrete-headings/
+
+A discrete heading also will not get a section number in the Customer Portal build of the doc. Previously, we would use plain bold mark-up around a heading like this, but discrete headings also allow you to ignore section nesting rules (like jumping from a `==` section level to a `====` level if you wanted for some style reason).
```
Hmm probably not, hadn't considered. Tho now that I re-read, I do kind of like that it discourages "plain bold mark-up" as a fake heading, in case someone thought of that on their own later.

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -287,98 +287,137 @@ Do not use backticks or other markup in assembly or module headings.
 ----
 [role="_additional-resources"]
-[id="additional-resources_configuring-alert-notifications"]
+[id="additional-resources_{context"]
```
Whoops

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -287,98 +287,137 @@ Do not use backticks or other markup in assembly or module headings.
 ====
 
-The following examples include IDs that are unique to the "Configuring alert notifications" assembly:
+You can use the `{context}` variable to automatically match the value assigned to the `:context:` attribute in the assembly. Alternatively, you manually set the string after the `_`, provided that the full string that you form is unique across the repo.
```
Hm I thought from the Slack thread that we just wanted to prefer `_{context}` but not necessarily explicitly forbid manual strings. I'll re-read it.

adellape replied:
Eh, I'm just gonna lose this "Alternative" option, I can't think of a decent reason to provide it (considering these 3 prefixes _should_ only exist once each per assembly).

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -287,98 +287,137 @@ Do not use backticks or other markup in assembly or module headings.
+= Configuring alert notifications <2>
+----
+<1> Anchor ID
+<2> Title / section heading
```
Looks like spaceless is more common in the file, dunno if there's an actual guideline. But I've synced all of mine up as spaceless, and changed a few older instances in the file to be spaceless as well.

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -287,98 +287,137 @@ Do not use backticks or other markup in assembly or module headings.
+==== Anchoring in assembly files
 
-=== Anchoring in assembly files
+===== Assembly titles (Level 0 headings)
```
It's mentioned in callout 5 of "Assembly file metadata", but I'll add a mention around here too.
